src/ast_sel_cmp.cpp

```diff
@@ -147,34 +147,17 @@ namespace Sass {
 
   bool Selector_List::operator== (const Selector_List& rhs) const
   {
-    // for array access
-    size_t i = 0, n = 0;
-    size_t iL = length();
-    size_t nL = rhs.length();
-    // create temporary vectors to sort them for compare
-    std::vector<Complex_Selector_Obj> l_lst = this->elements();
-    std::vector<Complex_Selector_Obj> r_lst = rhs.elements();
-    std::sort(l_lst.begin(), l_lst.end(), OrderNodes());
-    std::sort(r_lst.begin(), r_lst.end(), OrderNodes());
-    // process loop
-    while (true)
-    {
-      // first check for valid index
-      if (i == iL) return iL == nL;
-      else if (n == nL) return iL == nL;
-      // access the vector items
-      Complex_Selector_Ptr l = l_lst[i];
-      Complex_Selector_Ptr r = r_lst[n];
-      // skip nulls
-      if (!l) ++i;
-      else if (!r) ++n;
-      // do the check
-      else if (*l != *r) break;
-      // advance
-      ++i; ++n;
+    if (&rhs == this) return true;
+    if (rhs.length() != length()) return false;
+    std::unordered_set<const Complex_Selector *, HashPtr, ComparePtrs> lhs_set;
+    lhs_set.reserve(length());
+    for (const Complex_Selector_Obj &element : elements()) {
+      lhs_set.insert(element.ptr());
     }
-    // not equal
-    return false;
+    for (const Complex_Selector_Obj &element : rhs.elements()) {
+        if (lhs_set.find(element.ptr()) == lhs_set.end()) return false;
+    }
+    return true;
```

Comment:
@glebm looks like it got reverted again with the previous commit and added here again (as I thought, due to the rebasing). Please check again that it is included in the final result. Sorry about the mess. I will also keep an eye out that the spec you mentioned will pass after all the refactoring!

Reply:
CI still passing sass-spec/spec/core_functions/color/hsla/error/one_arg ...